### PR TITLE
Make TryGetConstantValue static

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
As part of the migration away from CheckInvocation, TryGetConstantValue will need to be made static and moved from Binder to BinderUtils.

This change is Part 1 of 2, only changing the code path for FirstNameNodes.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs [hunk 1]
```diff
@@ -847,5 +848,95 @@ public static BinderCheckTypeResult CheckBinaryOpCore(IErrorContainer errorConta
...
+                            return true;
+                        }
+                    }
+
+                    break;
+                case NodeKind.FirstName:
+                    // Possibly a non-qualified enum value
+                    var firstNameNode = node.AsFirstName();
+
+                    if (context.NameResolver.Lookup(firstNameNode.Ident.Name, out var firstNameInfo, NameLookupPreferences.None))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs:902]: >context.NameResolver.Lookup [](http://example.com/codeflow?start=24&length=27)

This was changed from:
var firstNameInfo = GetInfo(firstNameNode);

I think this is safe and behaves the same.

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs [hunk 2]
```diff
@@ -847,5 +848,95 @@ public static BinderCheckTypeResult CheckBinaryOpCore(IErrorContainer errorConta
...
+                        }
+                    }
+
+                    break;
+                case NodeKind.DottedName:
+                    // Possibly an enumeration
+                    var dottedNameNode = node.AsDottedName();
+                    if (dottedNameNode.Left.Kind == NodeKind.FirstName)
+                    {
+                        if (context.NameResolver.Lookup(dottedNameNode.Left.AsFirstName().Ident.Name, out var dottedNameInfo, NameLookupPreferences.None))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs:920]: >context.NameResolver.Lookup [](http://example.com/codeflow?start=28&length=27)

This was changed from:
if (Document.GlobalScope.TryGetNamedEnum(dottedNameNode.Left.AsFirstName().Ident.Name, out var enumType))

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs (hunk 2 above)
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs]: This change will be in a future PR.

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs (hunk 1 above)
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs]: This appears to be safe, as it's passing the Lab in PAClient

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 3]
```diff
@@ -1269,13 +1269,15 @@ public bool TryGetConstantValue(TexlNode node, out string nodeValue)
                 case NodeKind.FirstName:
                     // Possibly a non-qualified enum value
                     var firstNameNode = node.AsFirstName();
-                    var firstNameInfo = GetInfo(firstNameNode);
-                    if (firstNameInfo.Kind == BindKind.Enum)
+                    if (context.NameResolver.Lookup(firstNameNode.Ident.Name, out var firstNameInfo, NameLookupPreferences.None))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:1272]: >context.NameResolver [](http://example.com/codeflow?start=24&length=20)

This is an instance method on Binder? 
But also takes a check context... do we need that if we're an instance method?
Can we just get NameResolver from the binder?

Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:1272]: This is a partial transition to being static. The signature in PA client is also updated. However, the second code block below (for DottedNamedNode) still does not work. This partial change confirms which code block was causing the problem.

**Review by lesaltzm — APPROVED** [review]
🚢

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**